src/util/address.rs

```diff
@@ -180,6 +180,38 @@ impl Address {
         }
     }
 
+    /// Get an [Address] from an output script.
+    pub fn from_script(script: &script::Script, network: Network) -> Option<Address> {
+        Some(Address {
+            payload: if script.is_p2pkh() {
+                Payload::PubkeyHash(Hash::from_slice(&script.as_bytes()[3..23]).unwrap())
+            } else if script.is_p2sh() {
+                Payload::ScriptHash(Hash::from_slice(&script.as_bytes()[2..22]).unwrap())
+            } else if script.is_v0_p2wpkh() {
+                match bitcoin_bech32::WitnessProgram::new(
+                    bitcoin_bech32::u5::try_from_u8(0).expect("0<32"),
+                    script.as_bytes()[2..22].to_vec(),
+                    Address::bech_network(network),
+                ) {
+                    Ok(prog) => Payload::WitnessProgram(prog),
+                    Err(_) => return None,
+                }
+            } else if script.is_v0_p2wsh() {
```

Comment:
Is there a way we can support all segwit versions?

Reply:
You mean a generic thing without knowing if the witness program actually makes sense/is standard? That's dangerous no? But yeah I agree we could perhaps rename to `from_standard_script` and document clearly that it only supports standard scripts the implementation is aware of.

Reply:
It's not dangerous. It is explicitly recommended that any software that supports segwit addresses should support all versions (enforcing length constraints etc on versions that it understands).

Reply:
Hmm, https://github.com/bitcoin/bips/blob/master/bip-0173.mediawiki#Segwit_address_format is the only citation I can find. But I recall gmax saying many times that this was a design goal of the Segwit address format, so that new versions wouldn't cause the whole world to require updating all the time.

Reply:
"updating all the time" like segwit versions come like [what's the correct comparison in english?] pairs of trousers :D 

Perhaps an `is_standard` method or something than?

Reply:
F.e. I don't think blockstream.info would want to show addresses for non-(yet)-existing segwit programs. Even though they are just blackholed funds and probably won't occur on the mainchain.

Reply:
:( Yes, blockstream.info should show addresses for not-yet-defined segwit witness versions. If it does not that's a bug.

Reply:
Refering to IRC convo: the bitcoin_bech32 crate doesn't allow arbitrary segwit programs. https://github.com/rust-bitcoin/rust-bitcoin/pull/255 solves that.